### PR TITLE
[core] Minor cpp improvement

### PR DESCRIPTION
**Pull request by dentiny**
Some C++ improvements to our codebase.

### CODE REVIEW

File: src/ray/core_worker/transport/normal_task_submitter.cc [hunk 1]
```diff
@@ -333,7 +333,7 @@ void NormalTaskSubmitter::RequestNewWorkerIfNeeded(const SchedulingKey &scheduli
   // same TaskID to request a worker
   auto resource_spec_msg = scheduling_key_entry.resource_spec.GetMutableMessage();
   resource_spec_msg.set_task_id(TaskID::FromRandom(job_id_).Binary());
-  const TaskSpecification resource_spec = TaskSpecification(resource_spec_msg);
+  const TaskSpecification resource_spec = TaskSpecification(std::move(resource_spec_msg));
```
Comment (rynewang) [inline src/ray/core_worker/transport/normal_task_submitter.cc:336]: is it safe to move out of scheduling_key_entry ? do we need scheduling_key_entry to keep it?

Comment (dentiny) [reply src/ray/core_worker/transport/normal_task_submitter.cc:336]: Check L334, we made a copy.

File: src/ray/raylet/scheduling/cluster_task_manager.cc [hunk 2]
```diff
@@ -50,16 +50,21 @@ void ClusterTaskManager::QueueAndScheduleTask(
...
         send_reply_callback(Status::OK(), nullptr, nullptr);
       });
   const auto &scheduling_class = task.GetTaskSpecification().GetSchedulingClass();
   // If the scheduling class is infeasible, just add the work to the infeasible queue
   // directly.
-  if (infeasible_tasks_.count(scheduling_class) > 0) {
-    infeasible_tasks_[scheduling_class].push_back(work);
+  auto infeasible_tasks_iter = infeasible_tasks_.find(scheduling_class);
+  if (infeasible_tasks_iter != infeasible_tasks_.end()) {
+    iter->second.emplace_back(std::move(work));
```
Comment (rynewang) [inline src/ray/raylet/scheduling/cluster_task_manager.cc:65]: iter -> infeasible_tasks_iter ?

Comment (dentiny) [reply src/ray/raylet/scheduling/cluster_task_manager.cc]: ... yes, fixed.

**Review by rynewang — APPROVED** [review]

**Pull request merged**